### PR TITLE
Set to_json default to JSON lines

### PR DESCRIPTION
**Pull request by albertvillanova**
With this PR, the method `Dataset.to_json`:
- is added to the docs
- defaults to JSON lines

### CODE REVIEW

**Comment by stas00** [comment]
This is perfect, @albertvillanova - thank you! Tested it to work.

Might it be a good idea to document the args to `to_json`?

**Review by lhoestq — COMMENTED** [review]
Cool ! I agree it's more convenient this way

Same comment as Stas: let's add `orient` and `lines` parameters to the docstring.
And maybe update the test to make sure these parameters work as expected.

Regarding the progress bar, we can definitely use tqdm in the `for` loop in the `_write` method of the json writer.

File: src/datasets/arrow_dataset.py [hunk 1]
```diff
@@ -2765,17 +2765,28 @@ def to_json(
         batch_size: Optional[int] = None,
         **to_json_kwargs,
     ) -> int:
-        """Exports the dataset to JSON lines.
+        """Export the dataset to JSON or JSON Lines.
```
Comment (stas00) [inline src/datasets/arrow_dataset.py:2768]:
```suggestion
        """Export the dataset to JSON lines or JSON.
```

If the default is jsonl, may be listing it first?

**Review by stas00 — APPROVED** [review]
Fantastic - thank you, @albertvillanova

**Comment by albertvillanova** [comment]
@lhoestq I added tests for both `lines` and `orient`.

**Review by lhoestq — APPROVED** [review]
Thanks !

**Pull request merged**